tests/bor/bor_reorg_test.go

```diff
@@ -276,3 +278,155 @@ func TestValidatorWentOffline(t *testing.T) {
 	assert.Equal(t, authorVal1, nodes[0].AccountManager().Accounts()[0])
 
 }
+
+func TestForkWithBlockTime(t *testing.T) {
+
+	cases := []struct {
+		sprint        uint64
+		blockTime     map[string]uint64
+		change        uint64
+		producerDelay uint64
+		forkExpected  bool
+	}{
+		{
+			sprint: 128,
+			blockTime: map[string]uint64{
+				"0":   5,
+				"128": 2,
+				"256": 8,
+			},
+			change:        2,
+			producerDelay: 8,
+			forkExpected:  false,
+		},
+		{
+			sprint: 64,
+			blockTime: map[string]uint64{
+				"0":  5,
+				"64": 2,
+			},
+			change:        1,
+			producerDelay: 5,
+			forkExpected:  false,
+		},
+		{
+			sprint: 16,
+			blockTime: map[string]uint64{
+				"0":  2,
+				"64": 5,
+			},
+			change:        4,
+			producerDelay: 4,
+			forkExpected:  true,
+		},
+	}
+	log.Root().SetHandler(log.LvlFilterHandler(log.LvlInfo, log.StreamHandler(os.Stderr, log.TerminalFormat(true))))
+	fdlimit.Raise(2048)
+
+	// Generate a batch of accounts to seal and fund with
+	faucets := make([]*ecdsa.PrivateKey, 128)
+	for i := 0; i < len(faucets); i++ {
+		faucets[i], _ = crypto.GenerateKey()
+	}
+
+	// Create an Ethash network based off of the Ropsten config
+	genesis := initGenesis(t, faucets)
+
+	for _, test := range cases {
+		genesis.Config.Bor.Sprint = test.sprint
+		genesis.Config.Bor.Period = test.blockTime
+		genesis.Config.Bor.BackupMultiplier = test.blockTime
+		genesis.Config.Bor.ProducerDelay = test.producerDelay
+
+		var (
+			stacks []*node.Node
+			nodes  []*eth.Ethereum
+			enodes []*enode.Node
+		)
+
+		for i := 0; i < 2; i++ {
```

Comment:
Why 2?

Reply:
For the first, we are waiting for those many blocks to get mined before doing the checks.
And there are 2 nodes in the network mining